maintenance_plan/models/maintenance_equipment.py

```diff
@@ -102,10 +102,10 @@ def _cron_generate_requests(self):
             Generates maintenance request on the next_maintenance_date or
             today if none exists
         """
-        for plan in self.env['maintenance.plan'].search(
+        for plan in self.env['maintenance.plan'].sudo().search(
```

Comment:
but this will do the search as the admin user, so it will search in all companies, right?

Reply:
It's right. That code belogs to preventive requests creation task, run by OdooBot user, and this user shouldn't have access to every company, once plan becomes multi-company, so `sudo()` is needed.